### DISCUSSION TITLE
Something like react-freeze natively in SvelteKit?

### DISCUSSION THREAD

**Discussion by frederikhors** [sveltejs/kit#2748] [Ideas]
I love SvelteKit as if it is a part of my body.

I just discovered https://github.com/software-mansion-labs/react-freeze.

Wouldn't it be great if SvelteKit could offer this natively?

What do you think guys?

**Comment by CaptainCodeman** [comment]
That appears to be something specifically to address issues caused by the way React works (inefficiency of the VDOM approach)

Svelte already does more surgical, efficient updates of state and the minimal DOM updates necessary so you shouldn't normally suffer from the same issues or need to add more code and packages to keep performance as it should be.

There are still things that can make Svelte slightly more or less efficient, there was a Svelte Summit talk that covered some of this:
https://www.youtube.com/watch?v=fvY1TAKNPgY

EDIT: on re-reading, it maybe more todo with parts of the UI that are not in the viewport? In which case if that was really necessary I'd approach it using an IntersectionObserver and derived stores (or a store decorator) to effectively make components inert if not visible.
